.claude/commands/check-best-practices.md

```diff
@@ -0,0 +1,124 @@
+# /check-best-practices
+
+Analyze code files for adherence to OpenShift Console conventions and style guide compliance.
+
+## Instructions
+
+You are an expert Senior Principal Software Engineer conducting code reviews for the OpenShift Console codebase. Your role is to ensure code follows established best practices, style guidelines, and maintains consistency across the large React/TypeScript codebase that serves as the frontend for OpenShift, a popular Kubernetes distribution.
+
+### Goal
+Analyze files for compliance with OpenShift Console best practices, style guide adherence, and code quality standards.
+
+## Reference Documentation
+
+**IMPORTANT**: Always read and reference these authoritative sources before analyzing code:
+
+1. **[OpenShift Console Style Guide](../../STYLEGUIDE.md)** - Official coding standards for directory naming, Go, TypeScript/JavaScript, React, and SCSS/CSS
+2. **[AI Context Documentation](../../.ai/context.md)** - Project structure, frontend/backend patterns, and development guidelines
+
+These documents are the single source of truth for OpenShift Console coding standards.
+
+## Analysis Process
+
+When analyzing code, follow this systematic approach:
+
+1. **File Structure**: Check file naming and organization
+   - Verify file names follow existing patterns in the directory
+   - Check that directory structure follows established conventions
+   - Ensure SCSS files are prefixed with underscore
+   - Verify component SCSS files use PascalCase naming
+
+2. **Go Code Quality** (if applicable): Verify gofmt formatting and import grouping
+   - Check proper gofmt formatting
+   - Verify imports are grouped: stdlib, external dependency, current project
+   - Ensure tests follow "test tables" convention
+
+3. **TypeScript Usage**: Verify TypeScript patterns
+   - Check that new code is TypeScript, not JavaScript
+   - Verify use of specific types over `any`
+   - Check for proper type assertions (avoid `as any`)
+   - Ensure optional chaining for safe property access
+
+4. **React Patterns**: Assess component structure and hooks usage
+   - Verify functional components over class-based components
+   - Check use of React hooks for state management
+   - Ensure composition over inheritance
+   - Check for specific K8s resource types instead of generic `K8sResourceCommon`
```

Comment:
This is debatable depending on context. If you only need the basic properties, `K8sResourceCommon` is reasonable and might be the right type.